### DISCUSSION TITLE
Code Styling guidelines for the backend

### DISCUSSION THREAD

**Discussion by silentninja** [mathesar-foundation/mathesar#1007] [Maintainer discussions]
In order to maintain a consistent codebase, it would be helpful that we agree upon a certain styleguide and maybe few tools that could aid us in following those styleguide.

**Comment by dmos62** [comment]
There's a frustration I've been experiencing about long lines of code. I often find myself wanting to shorten lines in our code. We often go beyond 80 characters, and it's not uncommon to see lines beyond 100 characters. For me, longer lines add to cognitive complexity; my eyes read best when there's less horizontal intensity and more vertical intensity. Also, viewing two files side-by-side gets unergonomic when you have very long lines.

I'd love for us to setup some automatic tooling for controlling vertical vs. horizontal intensity.

**Comment by seancolsen** [comment]
Others probably saw this HN thread too: [Django: Reformatted code with Black](https://news.ycombinator.com/item?id=30258530). From the top comment:

> I suspect there are lots of other both open source and private projects that are also making the change now. This is a show of confidence in Black as the standard code formatter for Python.

Here's my take:

- Code which is styled consistently across one project is easier/faster to **read**. I care much more about code stye consistency than I care about the specific styling choices.

- Using a tool to auto-format code increases the speed with which I **write** code. I love having my editor set up to auto-format code on save and on paste. It really puts me into a different mindset. We recently [set up Prettier](https://github.com/centerofci/mathesar/pull/861) on the front end and it's been great.

- Opinionated auto-formatters like Black and Prettier make the code consistent -- not just across one project -- but across a larger ecosystem, making it easier for newcomers to read the code. I like that.

- I don't always like how Black formats code. Sometimes it uses up a lot of lines for deeply nested expressions. But also that ugly formatting often encourages me to restructure my code by lifting expressions into more descriptive variables. And occasionally I'll tell Black to shut up bu using `# fmt: off`.

**All-in-all this comment serves as a gentle vote for Black, but a strong vote for setting up _some_ kind of auto-formatting tooling.**

**Reply by dmos62** [reply]
I strongly second you on all points, except that I've not experienced how Black formats code, though I read its style guide and I liked it.
